app/dashboard/static/js/app/view-tests-job-branch-kernel-plan.2020.1.js

```diff
@@ -0,0 +1,597 @@
+        var container;
+        var node;
+
+        container = document.getElementById('accordion-container');
+        node = document.createElement('div')
+        node.className = 'pull-center';
+        node.appendChild(document.createTextNode('No test data available'));
+        html.replaceContent(container, node);
+    }
+
+    function updateRuns(results) {
+        gPanel = testView(results, gFileServer);
+        gPanel.draw();
+    }
+
+    function createLabResultsCount(total, pass, fail, unknown) {
+        var docFrag;
+        var smallNode;
+        var spanNode;
+        var tooltipNode;
+
+        docFrag = document.createDocumentFragment();
+        tooltipNode = docFrag.appendChild(html.tooltip());
+        html.addClass(tooltipNode, 'default-cursor');
+        tooltipNode.title =
+            "Total, passed, regressions and unknown test results for this lab";
+
+        smallNode = tooltipNode.appendChild(document.createElement('small'));
+        smallNode.appendChild(document.createTextNode(
+            '(' + format.number(total)));
+        smallNode.insertAdjacentHTML('beforeend', '&nbsp;&mdash;&nbsp;');
+
+        spanNode = smallNode.appendChild(document.createElement('span'));
+        spanNode.className = 'green-font';
+        spanNode.appendChild(document.createTextNode(format.number(pass)));
+
+        smallNode.insertAdjacentHTML('beforeend', '&nbsp;/&nbsp;');
+
+        spanNode = smallNode.appendChild(document.createElement('span'));
+        spanNode.className = 'red-font';
+        spanNode.appendChild(document.createTextNode(format.number(fail)));
+
+        smallNode.insertAdjacentHTML('beforeend', '&nbsp;/&nbsp;');
+
+        spanNode = smallNode.appendChild(document.createElement('span'));
+        spanNode.className = 'yellow-font';
+        spanNode.appendChild(
+            document.createTextNode(format.number(unknown)));
+
+        smallNode.appendChild(document.createTextNode(')'));
+
+        return docFrag;
+    }
+
+    function updateLabs(results) {
+        var labResults = {};
+
+        function initLab(lab) {
+            labResults[lab] = {};
+        }
+        Object.keys(gPanel.allLabs).forEach(initLab);
+
+        function mapResults(data) {
+            var lab = data.operation_id[0];
+            var result = data.operation_id[1];
+            var count = data.result[0].count;
+            labResults[lab][result] = count
+        }
+        results.forEach(mapResults);
+
+        function updateLab(lab) {
+            var res = labResults[lab];
+
+            html.replaceContent(
+                document.getElementById('test-count-' + lab),
+                createLabResultsCount(res['total'], res['success'],
+                                      res['regressions'], res['unknown'])
+            );
+        }
+        Object.keys(gPanel.allLabs).forEach(updateLab);
+    }
+
+    function updateRegressions(results) {
+        function parseBatchData(data) {
```

Comment:
@gctucker you've defined two function scopes inside another on (`updateRegressions`). Any special reason? Maybe we could split it in 3 functions, as steps to have regressions updated and remove `updateRegressions`.

Reply:
We can't remove `updateRegressions`, it's the function that updates the regressions as per its name and calls the main loop:
```
results.forEach(parseBatchData);
```
Then the nested functions are there because they are only useful within the context of `parseBatchData`.  The can be moved outside of it and have a wider scope but I don't see the benefit.